permissions.go

```diff
@@ -0,0 +1,78 @@
+package stream_chat //nolint: golint
+
+import (
+	"net/http"
+	"path"
+	"time"
+)
+
+type CustomPermission struct {
+	Id          string                 `json:"id"`
+	Name        string                 `json:"name"`
+	Description string                 `json:"description"`
+	Action      string                 `json:"action"`
+	Owner       bool                   `json:"owner"`
+	SameTeam    bool                   `json:"same_team"`
+	Condition   map[string]interface{} `json:"condition"`
+	Custom      bool                   `json:"custom"`
+	Level       string                 `json:"level"`
+}
+
+type getPermissionResponse struct {
+	Permission *CustomPermission `json:"permission"`
+}
+
+type listPermissionsResponse struct {
+	Permissions []*CustomPermission `json:"permissions"`
+}
+
+type Role struct {
+	Name      string    `json:"name"`
+	Custom    bool      `json:"custom"`
+	Scopes    []string  `json:"scoped"`
+	CreatedAt time.Time `json:"created_at"`
+	UpdatedAt time.Time `json:"updated_at"`
+}
+
+type roleResponse struct {
+	Roles []*Role `json:"roles"`
+}
+
+type Permissions struct {
+	client *Client
+}
```

Comment:
I don't like the name `Permissions` in a plural form but `Permission` is already used....

Reply:
`PermissionClient` ?

Reply:
Actually it makes me think we could probably move each entity to its own package.

Reply:
✅